src/internet_identity/src/activity_stats/event_stats/tests.rs

```diff
@@ -0,0 +1,364 @@
+            ii_domain: Some(IIDomain::Ic0AppDomain),
+            frontend: EXAMPLE_URL.to_string(),
+            session_duration_ns: Duration::from_secs(1800).as_nanos() as u64,
+        }),
+    };
+    update_events_internal(event1, TIMESTAMP, &mut storage);
+
+    assert_eq!(
+        storage
+            .event_aggregations
+            .get(&"PD_sess_sec_24h_ic0.app_https://example.com".to_string())
+            .unwrap(),
+        1800
+    );
+
+    let event2 = EventData {
+        event: Event::PrepareDelegation(PrepareDelegationEvent {
+            ii_domain: Some(IIDomain::Ic0AppDomain),
+            frontend: EXAMPLE_URL.to_string(),
+            session_duration_ns: Duration::from_secs(900).as_nanos() as u64,
+        }),
+    };
+    update_events_internal(event2, TIMESTAMP + 1, &mut storage);
+
+    assert_eq!(
+        storage
+            .event_aggregations
+            .get(&"PD_sess_sec_24h_ic0.app_https://example.com".to_string())
+            .unwrap(),
+        2700
+    );
+
+    let event3 = EventData {
+        event: Event::PrepareDelegation(PrepareDelegationEvent {
+            ii_domain: Some(IIDomain::Ic0AppDomain),
+            frontend: EXAMPLE_URL.to_string(),
+            session_duration_ns: Duration::from_secs(1).as_nanos() as u64,
+        }),
+    };
+    update_events_internal(event3, TIMESTAMP + 1 + DAY_NS, &mut storage);
+
+    assert_eq!(
+        storage
+            .event_aggregations
+            .get(&"PD_sess_sec_24h_ic0.app_https://example.com".to_string())
+            .unwrap(),
+        1
+    );
+}
+
+#[test]
+fn should_remove_zero_weighted_aggregations() {
+    let mut storage = test_storage();
+    let event = EventData {
+        event: Event::PrepareDelegation(PrepareDelegationEvent {
+            ii_domain: Some(IIDomain::Ic0AppDomain),
+            frontend: EXAMPLE_URL.to_string(),
+            session_duration_ns: Duration::from_secs(900).as_nanos() as u64,
+        }),
+    };
+
+    update_events_internal(event, TIMESTAMP, &mut storage);
+
+    assert_eq!(storage.event_aggregations.len(), 4);
+    const EXPECTED_AGGREGATIONS: [&str; 4] = [
+        "PD_count_24h_ic0.app_https://example.com",
+        "PD_sess_sec_24h_ic0.app_https://example.com",
+        "PD_count_30d_ic0.app_https://example.com",
+        "PD_sess_sec_24h_ic0.app_https://example.com",
+    ];
+    for key in EXPECTED_AGGREGATIONS.iter() {
+        assert!(storage.event_aggregations.contains_key(&key.to_string()));
+    }
+
+    let event2 = EventData {
+        event: Event::PrepareDelegation(PrepareDelegationEvent {
+            ii_domain: Some(IIDomain::Ic0AppDomain),
+            frontend: EXAMPLE_URL_2.to_string(),
+            session_duration_ns: Duration::from_secs(900).as_nanos() as u64,
+        }),
+    };
+
+    // after this update, the previous aggregations should be removed as their weight is 0
+    update_events_internal(event2, TIMESTAMP + 30 * DAY_NS, &mut storage);
```

Comment:
Should this 30 be part of the name of the function maybe? To check that events are pruned after 30 days.